src/domain.ts

```diff
@@ -73,16 +73,29 @@ async function setSlidesSettings(editor: Editor, repository: Repository) {
 import { settings as defaults } from "./settings";
 
 function getSlidesSettings(editor: Editor): Settings {
-  const { theme, fontFamily, previewMarkdownFiles } = editor.getConfiguration();
-
-  return JSON.stringify({
-    ...defaults,
-    ...(theme && { "workbench.colorTheme": theme }),
-    ...(fontFamily && { "editor.fontFamily": fontFamily }),
-    ...(fontFamily && { "terminal.integrated.fontFamily": fontFamily }),
-    ...(previewMarkdownFiles && {
+  const config = editor.getConfiguration();
+  const stdConfig = {
+    ...(config.theme && { "workbench.colorTheme": config.theme }),
+    ...(config.fontFamily && { "editor.fontFamily": config.fontFamily }),
+    ...(config.fontFamily && {
+      "terminal.integrated.fontFamily": config.fontFamily
+    }),
+    ...(config.previewMarkdownFiles && {
       "slides.previewMarkdownFiles": true
     })
+  };
+  const extendedConfig = {
+    ...config
+  };
+  // Pruning standardConfig stuff so we don't get unnecessary entries in the
+  // merged object.
+  delete extendedConfig.theme;
+  delete extendedConfig.fontFamily;
+  delete extendedConfig.previewMarkdownFiles;
```

Comment:
I'll try to clarify the code since it's not super clear what the different config stands for (std vs. defaults vs. extended), nor why we would remove some keys from the "extended config" and not the others.